intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -1970,3 +1973,113 @@ def convert_date(value):
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('source.reverse_dns', 'hostname'),
+        ('destination.ip', 'dst_ip'),
+        ('destination.port', 'dst_port'),
+        ('destination.asn', 'dst_asn'),
+        ('destination.geolocation.cc', 'dst_geo'),
+        ('destination.fqdn', 'dst_dns', validate_fqdn),
+        ('extra.', 'type', validate_to_none),
+        ('extra.', 'naics', invalidate_zero),
+        ('extra.', 'sic', invalidate_zero),
+        ('extra.', 'sector', validate_to_none),
+        ('extra.destination.sector', 'dst_sector', validate_to_none),
+        ('extra.', 'public_source', validate_to_none),
+        ('extra.', 'sensorid', validate_to_none),
+        ('extra.', 'pattern', validate_to_none),
+        ('extra.', 'url', validate_to_none),
+        ('extra.file.md5', 'file_md5', validate_to_none),
+        ('extra.file.sha256', 'file_sha256', validate_to_none),
+        ('extra.', 'request_raw', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.taxonomy': 'information gathering',
+        'classification.type': 'scanner',
+        'classification.identifier': 'http-scanners',
+    }
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/ICS-Scanners
+ics_scanners = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('source.reverse_dns', 'hostname'),
+        ('protocol.application', 'protocol'),
+        ('destination.ip', 'dst_ip'),
+        ('destination.port', 'dst_port'),
+        ('destination.asn', 'dst_asn'),
+        ('destination.geolocation.cc', 'dst_geo'),
+        ('destination.fqdn', 'dst_dns', validate_fqdn),
+        ('extra.', 'type', validate_to_none),
+        ('extra.', 'naics', invalidate_zero),
+        ('extra.', 'sic', invalidate_zero),
+        ('extra.', 'sector', validate_to_none),
+        ('extra.destination.sector', 'dst_sector', validate_to_none),
+        ('extra.', 'public_source', validate_to_none),
+        ('extra.', 'sensorid', validate_to_none),
+        ('extra.', 'state', validate_to_none),
+        ('extra.', 'slave_id', validate_to_none),
+        ('extra.', 'function_code', convert_int),
+        ('extra.', 'request', validate_to_none),
+        ('extra.', 'response', convert_int),
+    ],
+    'constant_fields': {
+        'classification.taxonomy': 'information gathering',
+        'classification.type': 'scanner',
+        'classification.identifier': 'ics-scanners',
```

Comment:
I'd actually use the protocol here again. Or is it not available for all rows?

Reply:
In the Shadowserver reports, the value of "protocol" usually specifies the _transport_ protocol (tcp or udp). However, in the "ICS Scanners" report, the value of "protocol" reflects the _application_ protocol (e.g. http or s7comm).
I'll ask Shadowserver if the format of the ICS Scanners report can be changed so that protocol specified the transport protocol there as well and an an additional column "application" can be added for the application protocol.